sklearn/datasets/kddcup99.py

```diff
@@ -0,0 +1,414 @@
+
+https://archive.ics.uci.edu/ml/machine-learning-databases/kddcup99-mld/kddcup.data.gz
+
+"""
+
+import sys
+import errno
+from gzip import GzipFile
+from io import BytesIO
+import logging
+import os
+from os.path import exists, join
+try:
+    from urllib2 import urlopen
+except ImportError:
+    from urllib.request import urlopen
+
+import numpy as np
+
+from .base import get_data_home
+from .base import Bunch
+from .mldata import fetch_mldata
+from ..externals import joblib
+from ..utils import check_random_state
+
+
+URL = ('http://archive.ics.uci.edu/ml/'
+       'machine-learning-databases/kddcup99-mld/kddcup.data_10_percent.gz')
+
+# URL = ('http://archive.ics.uci.edu/ml/'
+#        'machine-learning-databases/kddcup99-mld/kddcup.data.gz')
+
+
+logger = logging.getLogger()
+
+
+def load_kddcup99(subset=None, shuffle=False, random_state=None):
+    """Load and return the kddcup 99 dataset (regression).
+
+    The KDD Cup '99 dataset was created by processing the tcpdump portions
+    of the 1998 DARPA Intrusion Detection System (IDS) Evaluation dataset,
+    created by MIT Lincoln Lab [1] . The artificial data was generated using
+    a closed network and hand-injected attacks to produce a large number of
+    different types of attack with normal activity in the background.
+    As the initial goal was to produce a large training set for supervised
+    learning algorithms, there is a large proportion (80.1%) of abnormal
+    data which is unrealistic in real world, and inapropriate for unsupervised
+    anomaly detection which aims at detecting 'abnormal' data, ie
+    1) qualitatively different from normal data
+    2) in large minority among the observations.
+    We thus transform the KDD Data set into two differents data set: SA and SF.
+    -SA is obtained by simply selecting all the normal data, and a small
+    proportion of abnormal data to gives an anomaly proportion of 1%.
+    -SF is obtained as in [2]
+    by simply picking up the data whose attribute logged_in is positive, thus
+    focusing on the intrusion attack, which gives a proportion of 0.3% of
+    attack.
+    -http and smtp are two subsets of SF corresponding with third feature
+    equal to 'http' (resp. to 'smtp')
+
+    [1] : Analysis and Results of the 1999 DARPA Off-Line Intrusion Detection
+    Evaluation Richard Lippmann, Joshua W. Haines, David J. Fried, Jonathan 
+    Korba, Kumar Das 
+
+    [2] A Geometric Framework for Unsupervised Anomaly Detection: Detecting 
+    Intrusions in Unlabeled Data (2002) by Eleazar Eskin , Andrew Arnold , 
+    Michael Prerau , Leonid Portnoy , Sal Stolfo 
+
+    General KDD structure :
+    ==============      ==================
+    Samples total       4898431
+    Dimensionality      41
+    Features            discrete (int) or continuous (float)
+    Targets             str, 'normal' or name of the anomaly type
+    ==============      ==================
```

Comment:
this will not render properly in the doc. The number of = should be equal to field length.